src/frontend/sharedComponents/BGMapCard.tsx

```diff
@@ -0,0 +1,130 @@
+import * as React from "react";
+import { defineMessages, useIntl } from "react-intl";
+import { StyleSheet, View, Text } from "react-native";
+import { TouchableOpacity } from "react-native-gesture-handler";
+import MapboxGL from "@react-native-mapbox-gl/maps";
+import { StackNavigationProp } from "react-navigation-stack/lib/typescript/src/vendor/types";
+
+import { LIGHT_GREY, MEDIUM_GREY } from "../lib/styles";
+import { ViewStyleProp } from "../sharedTypes";
+import { Pill } from "./Pill";
+import LocationContext from "../context/LocationContext";
+
+const m = defineMessages({
+  currentMap: {
+    id: "sharedComponents.BGMapCard.currentMap",
+    defaultMessage: "Current Map",
+  },
+  abbrevMegabyte: {
+    id: "sharedComponents.BGMapCard.abbrevMegabyte",
+    defaultMessage: "MB",
+    description: "The abbreviation for megabyte",
+  },
+});
+
+//ToDo: API calls to get styleURL, zoom level, center coordinate, etc.
+
+interface BGMapCardProps {
+  mapId: string;
+  mapTitle: string;
+  mapSize: number;
+  navigation: StackNavigationProp;
+  style?: ViewStyleProp;
+  onPress?: (() => void) | null;
+}
+
+export const BGMapCard = ({
+  mapSize,
+  mapTitle,
+  style,
+  onPress,
+  mapId,
+  navigation,
+}: BGMapCardProps) => {
+  const { formatMessage: t } = useIntl();
+  const { navigate } = navigation;
+  const { position } = React.useContext(LocationContext);
+  const [styleUrl, setStyleUrl] = React.useState<string>(
+    "mapbox://styles/mapbox/streets-v11"
+  );
+  const [zoomLevel, setZoomLevel] = React.useState<number>(6);
+  const [centerCoordinate, setCenterCoordinate] = React.useState<
+    (number | undefined)[]
+  >([position?.coords.longitude, position?.coords.latitude]);
+
+  function onPressDefault() {
+    navigate("OfflineAreas", { mapId });
+  }
+
+  React.useEffect(() => {
+    function getStyleURL() {
+      //To do: API call to get styleURL
+      return MapboxGL.StyleURL.Street;
+    }
+    function getZoomLevel() {
+      //To do: API call to get zoom level
+      //This should be min zoom. Where is this coming from?
+      return 6;
+    }
+    function getCenterCoordinate() {
+      //To do: API call to get center coordinate
+      //I believe this should come from the styleJSON `center` property
+      return [position?.coords.longitude, position?.coords.latitude];
+    }
```

Comment:
with the mapserver, this will all be a single api call that retrieves the offline style, which contains the information about the zoom level and center coordinate too. You'd then pass this style into the `styleJSON` prop of the map

Reply:
thats fine, I think we can deal with this when the api is built!